apps/explorer/src/pages/transaction-result/programmable-transaction-view/utils.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { type SuiArgument } from '@mysten/sui.js';
+
+export const stringifyCommandData = (commandData: any): string => {
+    if (Array.isArray(commandData)) {
+        return commandData
+            .map((data: any) => {
+                if (typeof data === 'object') {
+                    return JSON.stringify(data);
+                }
+                return data;
+            })
+            .join(', ');
+    }
+    return JSON.stringify(commandData);
+};
+
+export function flattenSuiArguments(
+    data: (SuiArgument | SuiArgument[])[]
+): string {
+    if (!data) {
+        return '';
+    }
+
+    const res = [];
+    for (const value of data) {
+        if (typeof value === 'string') {
+            res.push(value);
+        } else if (Array.isArray(value)) {
+            res.push(`[${flattenSuiArguments(value)}]`);
+        } else if (typeof value === 'object') {
```

Comment:
Doing this generically is a little weird, I'd just hard-code the couple cases we have (result, nested result, gas, input)